core/trino-main/src/main/java/io/trino/cost/JoinStatsRule.java

```diff
@@ -180,47 +182,61 @@ private PlanNodeStatsEstimate filterByEquiJoinClauses(
+        // clause separately because stats estimates would be way off.
+        // Sort known estimates by selectivity
+        Queue<PlanNodeStatsEstimateWithClause> knownSortedEstimates = clauses.stream()
+                .map(clause -> {
+                    ComparisonExpression predicate = new ComparisonExpression(EQUAL, clause.getLeft().toSymbolReference(), clause.getRight().toSymbolReference());
+                    return new PlanNodeStatsEstimateWithClause(filterStatsCalculator.filterStats(stats, predicate, session, types), clause);
+                })
+                .filter(estimateWithClause -> !estimateWithClause.getEstimate().isOutputRowCountUnknown())
+                .sorted(comparingDouble(estimateWithClause -> estimateWithClause.getEstimate().getOutputRowCount()))
+                .collect(toCollection(LinkedList::new));
+
+        if (knownSortedEstimates.isEmpty()) {
+            return PlanNodeStatsEstimate.unknown();
         }
 
-        return result;
-    }
+        PlanNodeStatsEstimate combinedEstimate = knownSortedEstimates.poll().getEstimate();
+        double combinedSelectivity = combinedEstimate.getOutputRowCount() / stats.getOutputRowCount();
+        double combinedIndependenceFactor = 1.0;
+        double independenceFactor = getJoinMultiClauseIndependenceFactor(session);
+        // For independenceFactor = 0.75 and terms t1, t2, t3
+        // Combined selectivity = (t1 selectivity) * ((t2 selectivity) ^ 0.75) * ((t3 selectivity) ^ (0.75 * 0.75))
+        // independenceFactor = 1 implies the terms are assumed to have no correlation and their selectivities are multiplied without scaling.
+        // independenceFactor = 0 implies the terms are assumed to be fully correlated and only the most selective term drives the selectivity.
+        while (!knownSortedEstimates.isEmpty()) {
+            PlanNodeStatsEstimateWithClause estimateWithClause = knownSortedEstimates.poll();
+            PlanNodeStatsEstimate clauseEstimate = estimateWithClause.getEstimate();
+            EquiJoinClause clause = estimateWithClause.getClause();
+            combinedIndependenceFactor *= independenceFactor;
+            double scaledClauseSelectivity = Math.pow(clauseEstimate.getOutputRowCount() / stats.getOutputRowCount(), combinedIndependenceFactor);
+            // If scaledClauseSelectivity goes above UNKNOWN_FILTER_COEFFICIENT threshold,
+            // fall back to multiplying output row count by UNKNOWN_FILTER_COEFFICIENT
+            if (scaledClauseSelectivity >= UNKNOWN_FILTER_COEFFICIENT) {
+                knownSortedEstimates.add(estimateWithClause);
+                break;
+            }
+            combinedSelectivity *= scaledClauseSelectivity;
+            double outputRowCount = stats.getOutputRowCount() * combinedSelectivity;
+            combinedEstimate = filterByClause(combinedEstimate, clause, outputRowCount, types);
```

Comment:
as for filter, you could:
1. compute final selectivity (unified method between filter and join)
2. compute final range for symbols (different for filter and join)
3. apply final output row count on final ranges computed in 2)